packages/api/src/modules/grant/commonGrant.service.ts

```diff
@@ -0,0 +1,80 @@
+import { ApplicationDto, GrantDto, PaymentDto } from "@api-subventions-asso/dto";
+import providers from "../providers";
+import { JoinedRawGrant, RawGrant } from "./@types/rawGrant";
+import GrantProvider from "./@types/GrantProvider";
+
+export class CommonGrantService {
+    // not an adapter because it needs to call other modules' adapters
+
+    private readonly providerMap: { [providerId: string]: GrantProvider };
+
+    constructor() {
+        this.providerMap = {};
+        for (const provider of Object.values(providers as unknown as { [serviceName: string]: GrantProvider })) {
```

Comment:
J'aurai imaginé une méthode getGrandProvider() plutôt qu'utiliser un for of qui vérifie ça

Reply:
Je l'ai mis dans le constructeur pour pas recalculer à chaque appel